### DISCUSSION TITLE
Mimalloc引起ppocr推理崩溃

### DISCUSSION THREAD

**Discussion by xiaohan-W** [PaddlePaddle/PaddleOCR#12773] [Q&A]
端侧本地部署ppocr的推理程序。计划将ppocr编译为dll，并导出一个接口供外部调用。主程序ConsoleApplication1.exe调用这个导出的接口。主程序里引入mimalloc后，在推理过程中崩溃。

具体崩溃位置为检测过程的BoxesFromBitmap接口，见附图。

<img width="891" alt="Mimalloc引起PPOCR崩溃" src="https://github.com/PaddlePaddle/PaddleOCR/assets/82928602/1bac8a49-5145-43c0-b6dd-26d65bf9ec55">

其中主程序ConsoleApplication.exe的代码为
<img width="712" alt="ConsoleAppliction1Code" src="https://github.com/PaddlePaddle/PaddleOCR/assets/82928602/3c29ddce-898b-4b70-a654-731162b9bf98">

PPOCR推理库导出接口为
<img width="466" alt="PPOCR导出接口" src="https://github.com/PaddlePaddle/PaddleOCR/assets/82928602/b9ea5b01-1772-4bc9-a345-33717a19d69e">

请教可能是什么原因？是ppocr里内存管理可能有不恰当的地方，还是自身与mimalloc不兼容？是否与ppocr.dll的MT、MD模式有关？

**Comment by GreatV** [comment]
1. 尝试不使用mimalloc库,使用默认的内存分配器,看是否还会发生崩溃。
2. 检查ppocr.dll和ConsoleApplication1.exe的编译模式是否匹配,如果不匹配,需要重新编译使之一致。

**Comment by xiaohan-W** [comment]
> 1. 尝试不使用mimalloc库,使用默认的内存分配器,看是否还会发生崩溃。
> 2. 检查ppocr.dll和ConsoleApplication1.exe的编译模式是否匹配,如果不匹配,需要重新编译使之一致。

1. 不使用mimalloc库是不崩溃的。
2. 编译模式MD或MT对崩溃现象没有影响。修改为均为MT模式，引入mimalloc库崩溃，不引入的话不崩溃。

如果一定要引入mimalloc库的话有何解决办法？
